src/constructs/autoscaling/asg.ts

```diff
@@ -13,23 +13,24 @@ export interface GuAutoScalingGroupProps
   extends Omit<AutoScalingGroupProps, "imageId" | "osType" | "machineImage" | "instanceType" | "userData"> {
   instanceType?: InstanceType;
   imageId?: GuAmiParameter;
-  osType?: OperatingSystemType;
   machineImage?: MachineImage;
-  userData: string;
+  userData: UserData | string;
   securityGroups?: ISecurityGroup[];
   targetGroup?: ApplicationTargetGroup;
   overrideId?: boolean;
 }
 
 export class GuAutoScalingGroup extends AutoScalingGroup {
   constructor(scope: GuStack, id: string, props: GuAutoScalingGroupProps) {
+    const userData = props.userData instanceof UserData ? props.userData : UserData.custom(props.userData);
+
     // We need to override getImage() so that we can pass in the AMI as a parameter
     // Otherwise, MachineImage.lookup({ name: 'some str' }) would work as long
     // as the name is hard-coded
     function getImage(): MachineImageConfig {
       return {
-        osType: props.osType ?? OperatingSystemType.LINUX,
-        userData: UserData.custom(props.userData),
+        osType: OperatingSystemType.LINUX,
```

Comment:
Minor point: it might be worth mentioning this change in the PR description so that it's documented somewhere?

Reply:
Done.